.github/actions/publish-nuget/action.yml

```diff
@@ -60,7 +60,13 @@ runs:
         ${{ inputs.publish-version }}
         '@.Trim()
 
-        if ($version -eq 'USE_GITHUB_REF_NAME') { $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v") }
+        if ($version -eq 'USE_GITHUB_REF_NAME') {
+            $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
+            if (!($version -match '^\d+\.\d+\.\d+(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
+                Write-Output "Invalid version number! The correct format is vM.N.O or vM.N.O-alpha.X.[issue-code]."
```

Comment:
Should we use `{issue-code}` instead of `[issue-code]`? For people used to shell scripting [something] suggests that it's optional. We also use braces in string interpolation so it's a more natural fit.

Reply:
I was thinking that the issue code could be **optional**, that's why I uses square brackets in the issue description. But no-one asked for that, so let's not complicate things further.

Reply:
Braces hide the value in GHA output: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#masking-a-value-in-log.